File: iteration2/Models/Curriculum.java
package iteration2.Models;

import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.logging.FileHandler;
import java.util.logging.Logger;
import java.util.logging.SimpleFormatter;

public class Curriculum {

    private HashMap<Integer, ArrayList<Course>> courses = new HashMap<Integer, ArrayList<Course>>();

    public HashMap<Integer, ArrayList<Course>> getCourses() {
        return courses;
    }
    private static String file_path="/iteration2/Logs/Curriculum.log";
    private static Logger logger=Logger.getLogger(Curriculum.class.getName());
    private static FileHandler fileHandler;

    static {
        try {
            fileHandler = new FileHandler(System.getProperty("user.dir")+file_path,true);
            logger.addHandler(fileHandler);
            logger.setUseParentHandlers(false);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void customLog(boolean type,String message){
        SimpleFormatter formatter =new SimpleFormatter();
        fileHandler.setFormatter(formatter);
        if(type)
            logger.info(message);
        else
            logger.warning(message);
    }
    public boolean addCourseToSemester(Course course) {
        if(courses.get(course.getSemester()) == null) {
            courses.put(course.getSemester(), new ArrayList<>());
            courses.get(course.getSemester()).add(course);
            customLog(true, "Created new semester array list for courses and added courses to semester.");
        }
        else {
            courses.get(course.getSemester()).add(course);
            customLog(true, "Added courses to semester.");
        }
        return true;
    }
}

File: iteration3/Models/Curriculum.py
import logging
class Curriculum:
    def __init__(self, name, courses):
        self.__courses = courses
        self.__name = name

    logging.basicConfig(filename='Curriculum.log', filemode='w', format='%(name)s - %(levelname)s - %(message)s')
    @property
    def courses(self):
        return self.__courses

    @property
    def name(self):
        return self.__name

    def course_by_semester(self, semester_no):
        courses = []
        for course in self.__courses:
            if course.semester == semester_no:
                courses.append(course)
        return courses

    def custom_log(self, is_error, message):
        if is_error:
            self.logger.info(message)
        else:
            self.logger.warning(message)

    def add_course_to_semester(self, course):
        if course.get_semester() not in self.courses:
            self.courses[course.get_semester()] = []
            self.courses[course.get_semester()].append(course)
            self.custom_log(True, "Created new semester array list for courses and added courses to semester.")
        else:
            self.courses[course.get_semester()].append(course)
            self.custom_log(True, "Added courses to semester.")

        return True
